ts_src/psbt.ts

```diff
@@ -687,43 +706,47 @@ export class Psbt {
     const inputAgs: Buffer[] = [];
     const inputValues: string[] = [];
 
+    // counter for fetching blinding private key
+    let confidentialUtxoIndex = 0;
+
     // unblind inputs
     this.data.inputs.forEach((input: PsbtInput, index: number) => {
-      let prevout: any;
+      let prevout: WitnessUtxo;
       if (input.nonWitnessUtxo) {
         const prevTx = nonWitnessUtxoTxFromCache(c, input, index);
         const prevoutIndex = c.__TX.ins[index].index;
         prevout = prevTx.outs[prevoutIndex];
       } else {
-        prevout = { ...input.witnessUtxo };
+        prevout = input.witnessUtxo!;
       }
 
-      const unblindPrevout = {
-        value: '',
-        ag: Buffer.alloc(0),
-        abf: Buffer.alloc(0),
-        vbf: Buffer.alloc(0),
-      };
-      if (
-        prevout.rangeProof!.length > 0 &&
-        prevout.surjectionProof!.length > 0
-      ) {
-        const unblindProof = confidential.unblindOutput(
-          prevout.nonce,
-          blindingPrivkeys[index],
-          prevout.rangeProof!,
-          prevout.value,
-          prevout.asset,
-          prevout.script,
-        );
-        unblindPrevout.ag = unblindProof.asset;
-        unblindPrevout.value = unblindProof.value;
-        unblindPrevout.abf = unblindProof.assetBlindingFactor;
-        unblindPrevout.vbf = unblindProof.valueBlindingFactor;
+      let unblindPrevout: UnblindWitnessUtxoResult;
+
+      if (prevout.rangeProof != null && prevout.surjectionProof != null) {
+        const blindingPrivKey = blindingPrivkeys[confidentialUtxoIndex];
+        if (!blindingPrivKey) {
+          throw new Error(
+            'the number of blinding private keys does not match the number of confidential inputs',
+          );
+        }
```

Comment:
in go-elements, instead of implementing this logic, we opted for having the array of input blinding keys of the same length of the inputs.. in case an input is unconfidential, its key is null. This delegates the user to provide the correct list of blinding keys and the method to just check that in case an input is blinded, its binding key must not be null. 

I would keep the same logic, so for example if my tx has 3 inputs, the first unconf and the others conf, I would need to provide a list of keys like `[null, key1, key2]`.

Reply:
OK let's keep the go-elements logic :+1: